share/wake/lib/rust_wake/cargo.wake

```diff
@@ -0,0 +1,45 @@
+# Copyright 2023 SiFive, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You should have received a copy of LICENSE.Apache2 along with
+# this software. If not, you may obtain a copy at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package rust
+
+target findCargoPath Unit =
+    require Some cargo_home = getenv "CARGO_HOME"
+    else
+        require Some path = getenv "PATH"
+        else failWithError "neither $PATH nor $CARGO_HOME is set; cannot find cargo"
+
+        Pass (whichIn path "cargo")
```

Comment:
What does `whichIn` do when `cargo` is not on the `$PATH`?

Reply:
It just returns "cargo" which is the behavior that I was mimicking before. You wind up getting an error later on from execve